src/simulationarchive.c

```diff
@@ -609,6 +609,10 @@ void reb_simulationarchive_snapshot(struct reb_simulation* const r, const char*
                 fseek(of, field.size, SEEK_CUR);
             }while(field.type!=REB_BINARY_FIELD_TYPE_END && bytesread);
             long size_old = ftell(of);
+            
+            // store location after initial binary (including trailing blob) as first valid end of file
+            long last_blob=ftell(of) + sizeof(struct reb_simulationarchive_blob);
```

Comment:
This variable seems to get overwritten before its value is ever read.

Reply:
Good catch! This was left over from when I wrote the do while loop as a while loop...can definitely be removed!